e2e/test/deployment.ts

```diff
@@ -0,0 +1,202 @@
+import * as assert from "assert";
+import * as CodePush from "rest-definitions";
+import { Command } from "./utils/command";
+import { Error } from "./utils/error";
+import { makeRandomString } from "./utils/misc";
+import { Success } from "./utils/success";
+import { Validate } from "./utils/validate";
+var nixt = require("nixt");
+var tryJSON = require("try-json");
+
+export function deploymentTests() {
+    var appName: string = makeRandomString();
+    before((done: MochaDone) => {
+        nixt()
+            .stdout(Success.appAdd(appName))
+            .run(Command.appAdd(appName))
+            .end(done);
+    });
+    
+    after((done: MochaDone) => {
+        nixt()
+            .stdout(Success.appRm(appName))
+            .run(Command.appRm(appName))
+            .on(Command.PROMPT_ARE_YOU_SURE).respond(Command.RESPONSE_ACCEPT)
+            .end(done);
+    });
+    
+    describe("deployment ls", () => {
+        it("succeeds", (done: MochaDone) => {
+            nixt()
+                .expect(Validate.Deployments())
+                .run(Command.deploymentLs(appName))
+                .end(done);
+        });
+        
+        it("fails if app not found", (done: MochaDone) => {
+            var fakeAppName: string = "not_a_real_app";
+            
+            nixt()
+                .stderr(Error.appNotFound(fakeAppName))
+                .run(Command.deploymentLs(fakeAppName))
+                .end(done);
+        });
+        
+        it("fails with invalid format", (done: MochaDone) => {
+            nixt()
+                .stderr(Error.invalidFormat())
+                .run(Command.invalidFormat(Command.deploymentLs(appName)))
+                .end(done);
+        });
```

Comment:
what about checking the options for deployment command? for eg: we have `deployment ls <appName> -k` which returns the deployment keys, we should verify those too.